nautobot/extras/models/base_properties.py

```diff
@@ -0,0 +1,78 @@
+from django.contrib.contenttypes.models import ContentType
+from django.db import models
+from django.urls import reverse
+
+from nautobot.core.fields import AutoSlugField
+from nautobot.core.models import BaseModel
+from nautobot.extras.models.customfields import CustomFieldModel
+from nautobot.extras.models.change_logging import ChangeLoggedModel
+from nautobot.extras.models.relationships import RelationshipModel
+from nautobot.extras.models.mixins import NotesMixin
+from nautobot.utilities.choices import ColorChoices
+from nautobot.utilities.fields import ColorField
+from nautobot.utilities.querysets import RestrictedQuerySet
+
+
+class ContentTypeRelatedQuerySet(RestrictedQuerySet):
+    def get_for_model(self, model):
+        """
+        Return all `self.model` instances assigned to the given model.
+        """
+        content_type = ContentType.objects.get_for_model(model._meta.concrete_model)
+        return self.filter(content_types=content_type)
+
+    def get_by_natural_key(self, name):
+        return self.get(name=name)
+
+
+class BasePropertiesModel(BaseModel, ChangeLoggedModel, CustomFieldModel, RelationshipModel, NotesMixin):
```

Comment:
Any reason this can't be an `OrganizationalModel` subclass?

Reply:
status do not inherit from `DynamicGroupMixin` guess its same for role.

Reply:
Is there a reason it *shouldn't* inherit from DynamicGroupMixin?

Reply:
No there isn't a good reason. Status seems like it should/could certainly benefit from supporting dynamic groups, but we need to be cautious about the chicken and the egg problem as it relates to multiple inheritance here, in case there is one.